src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1607,6 +1624,15 @@ void Session::send_message()
     // not been initiated
     REALM_ASSERT(!m_unbind_message_sent);
 
+    const auto has_pending_test_command = std::any_of(m_pending_test_commands.begin(), m_pending_test_commands.end(),
+                                                      [](const PendingTestCommand& command) {
+                                                          return command.pending;
+                                                      });
+
+    if (has_pending_test_command) {
+        return send_test_command_message();
```

Comment:
This seems to be a session level message, so I think it should be sent after `bind` and `ident`